tinyasm/matinvert.cpp

```diff
@@ -3,8 +3,12 @@
 
 
 PetscErrorCode mymatinvert(PetscInt* n, PetscScalar* mat, PetscInt* piv, PetscInt* info, PetscScalar* work) {
-    PetscCallBLAS("LAPACKgetrf",LAPACKgetrf_(n,n,mat,n,piv,info));
-    PetscCallBLAS("LAPACKgetri",LAPACKgetri_(n,mat, n, piv,work,n,info));
+    PetscBLASInt *n_blas, *piv_blas, *info_blas;
+    PetscCall(PetscBLASIntCast(*n, n_blas));
+    PetscCall(PetscBLASIntCast(*piv, piv_blas));
+    PetscCall(PetscBLASIntCast(*info, info_blas));
```

Comment:
Should you be casting back to PETScInt after the blas calls?

Reply:
Info is an out parameter so should definitely be cast back.

piv is out for the factorisation but only in for the inverse so I don't think it's out for `mymatinvert`, so shouldn't need to be cast.

Reply:
I don't think so, I think this routine just checks that the cast is safe:

https://petsc.org/main/manual/blas-lapack/

Reply:
Actually it looks like the range of `PETScInt` is always equal or greater than the range of `PETScBLASInt` so you can probably get away with just assigning.

Reply:
I have changed the way we do this, I believe it to be resolved now